dnf5/commands/downgrade/downgrade.cpp

```diff
@@ -60,6 +60,9 @@ void DowngradeCommand::configure() {
     auto & context = get_context();
     context.set_load_system_repo(true);
     context.set_load_available_repos(Context::LoadAvailableRepos::ENABLED);
+    auto & config = context.base.get_config();
+    /// Allow dowgrade should be the default behavior of the command
+    config.get_allow_downgrade_option().set(libdnf::Option::Priority::COMMANDLINE, true);
```

Comment:
Just curious why did you use `COMMANDLINE` priority? Because what I'd do is plain `config.get_allow_downgrade_option().set(true);` (which defaults to RUNTIME).

Reply:
I would like to allow `--setopt` or option (e.g. `--allow-downgrade`) overrides. If both will use `runtime` priority then order will be important. I guess that configure section would override the cmd option and that I don't want.

Reply:
I see.

But I'm quite sure that `--setopt=allow_downgrade=0` will not work with your code, because you both use the same priority (`COMMANDLINE`), but your code is executed later: https://github.com/rpm-software-management/dnf5/blob/4447b5f467baa667399eb837e50bef0ddd3e1cb9/dnf5/main.cpp#L742

That means it will always override values set by commandline args parser which runs here: https://github.com/rpm-software-management/dnf5/blob/4447b5f467baa667399eb837e50bef0ddd3e1cb9/dnf5/main.cpp#L681

Is setting the `allow_downgrade` value for downgrade command even needed? The hard coded default is `true` anyway, and if you remove this line completely, `--setopt` should start working. Of course, the downgrade command will also follow values set in `dnf.conf` file, which is IMHO expected.

Reply:
This is tricky. What about to use PLUGINDEFAULT?  The only problem what I see is with DROPINCONFIG.

Reply:
Hm, this feels like misuse of `PLUGINDEFAULT` priority (at least judging by its name, I'm not completely sure what this priority was originally meant for).

What's the use-case you are trying to resolve? Situation when the user puts `allow_downgrade=false` to dnf.conf? 

Maybe what we need in first place is documentation of config options, where should be described what each option does. In this case, that option affects downgrading of **other then directly requested** installed packages during goal resolving. And if someone sets this option in dnf.conf, they should be aware (after reading the doc) that the `downgrade` command is inevitably affected. And for this case we should add the same `--allow-downgrade/--no-allow-downgrade` set of options.

Reply:
Ok, improvement of documentation is very good suggestion. But I am not sure whether it make sense to include it in this PR. Anyway I have to look to DNF4 code.

Reply:
Oh, sorry - I didn't mean that you should create a dnf.conf documentation here :) It's definitely a task for separate PR.